### PR TITLE
Add @_inlineable attribute

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test and merge

File: include/swift/AST/Attr.def [hunk 1]
```diff
@@ -164,9 +164,12 @@ SIMPLE_DECL_ATTR(nonobjc, NonObjC,
 SIMPLE_DECL_ATTR(_fixed_layout, FixedLayout,
                  OnVar | OnClass | OnStruct | OnEnum | UserInaccessible, 31)
 
+SIMPLE_DECL_ATTR(_fragile, Fragile,
+                 OnVar | OnSubscript | OnFunc | OnConstructor | UserInaccessible, 32)
```
Comment (DougGregor) [inline include/swift/AST/Attr.def]: You'll want to bump the module version...

**Comment by jrose-apple** [comment]
Please name this `@_inlineable`, even when prototyping. It doesn't apply to anything but decls with bodies, right?

Does this also belong on deinitializers?

File: include/swift/AST/Attr.def (hunk 1 above)
Comment (slavapestov) [inline include/swift/AST/Attr.def]: Good point.

**Comment by slavapestov** [comment]
@DougGregor I bumped the version. @jrose-apple I renamed the attribute and added a test that it works on deinits -- good catch.

**Comment by slavapestov** [comment]
@swift-ci Please test and merge

**Comment by jckarter** [comment]
It's also spelled `inlinable`, not `inlineable`.

**Comment by jrose-apple** [comment]
Oh no it isn't. :-) http://jrose-apple.github.io/swift-library-evolution/#inlineable-functions

(I went back and forth on it. I don't actually care what swift-evolution picks once we put this through review, which is probably the most dangerous thing to say.)

**Comment by slavapestov** [comment]
@swift-ci Please smoke test and merge

**Comment by slavapestov** [comment]
@swift-ci Please smoke test and merge

**Comment by slavapestov** [comment]
@jckarter @jrose-apple One of the reasons I wanted to go with `@_fragile` :-)

**Comment by jckarter** [comment]
We may as well head off an unnecessary swift-evolution bikeshed by using the correct spelling going in, or else picking a name whose spelling is less in doubt like Slava suggested.

**Pull request merged**